src/db/oracledb/members-dao.js

```diff
@@ -0,0 +1,205 @@
+import _ from 'lodash';
+
+import oracledb from 'oracledb';
+
+import { parseQuery } from 'utils/parse-query';
+
+import { getConnection } from './connection';
+
+import { generateWhereClause, generatePaginationParams, convertKeysToLowercase } from './general-dao';
+
+/**
+ * Return a list of members with pagination
+ *
+ * @param {object} query Query parameters
+ * @returns {Promise<object>[]} Promise object represents a list of members
+ */
+const getMembers = async (query) => {
+  const connection = await getConnection();
+  try {
+    const parsedQuery = parseQuery(query);
+    const whereClause = generateWhereClause(parsedQuery).toLowerCase();
+    const paginationParams = generatePaginationParams(parsedQuery);
+
+    const selectQuery = `
+      SELECT *
+      FROM library_api_members
+      ${whereClause}
+      OFFSET ${paginationParams.offset} ROWS FETCH NEXT ${paginationParams.perPage} ROWS ONLY
+    `;
+
+    const { rows } = await connection.execute(selectQuery);
+    return convertKeysToLowercase(rows);
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * Return a specific member by unique ID
+ *
+ * @param {string} id Unique member ID
+ * @returns {Promise<object>} Promise object represents a specific member or return undefined if
+ *                            not found
+ */
+const getMemberById = async (id) => {
+  const connection = await getConnection();
+  const query = `
+    SELECT *
+    FROM library_api_members
+    WHERE member_id = :memberId
+  `;
+
+  try {
+    const bindVars = { memberId: id };
+
+    const { rows } = await connection.execute(query, bindVars);
+    if (_.isEmpty(rows)) {
+      return undefined;
+    }
+    if (rows.length > 1) {
+      throw new Error('Expect a single object but got multiple results.');
+    } else {
+      const rawMembersLower = rows[0];
+      return convertKeysToLowercase([rawMembersLower])[0];
+    }
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * Update a specific member by unique ID
+ *
+ * @param {string} id Unique member ID
+ * @param {object} updateData Data to update
+ * @param {object} existingMember Exisitng member to update
+ * @returns {Promise<object>} Promise object represents the updated member or undefined if not found
+ */
+const updateMemberById = async (id, updateData, existingMember) => {
+  const connection = await getConnection();
+
+  try {
+    const updatedMemberData = {
+      ...existingMember,
+      ...updateData,
+    };
+
+    const updateQuery = `
+    UPDATE library_api_members
+    SET firstName = :firstName,
+        lastName = :lastName,
+        email = :email,
+        address = :address,
+        city = :city,
+        state = :state,
+        country = :country,
+        phoneNumber = :phoneNumber,
+        status = :status
```

Comment:
Are these all necessary fields for updating? For example, if you didn't provide any of these fields in your data packet, will you get any error or cause any expected behavior? You might want to review your openapi file again and see if you missed anything there. Feel free to ping me in Teams if you have more questions on this.

Reply:
Good catch, fixing these. I'll ask if anything is needed for more clarification.

Reply:
Btw firstly we retrieve if a record exists in the data based on GET/ID before changing it, and that retrieves everything for that record and in next step only the patched ones are updated in this case and working fine, until there is a case when no attributes are being passed in the Body then Lowercase() on that key value is giving an error, I'll still check into this case more deeply and fix this, but for now if attribute is not empty it works for any number of parameters passed.

Reply:
working correctly for all cases now.